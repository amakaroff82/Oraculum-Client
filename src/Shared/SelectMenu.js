import React from 'react';
import PropTypes from 'prop-types';
import { compose, withState, withHandlers } from 'recompose';
import Input from 'material-ui/Input';
import { MenuItem } from 'material-ui/Menu';
import Select from 'material-ui/Select';

export const SelectMenu = ({
  handleChange,
  InputId,
  InputProps,
  items,
  multiple,
  MenuProps,
  value,
}) => {
  const inputId = InputId || 'select-input';

  return (
    <Select
      input={<Input id={inputId} {...InputProps} />}
      MenuProps={MenuProps}
      multiple={multiple}
      onChange={handleChange}
      value={value}
    >
      {items &&
        items.map((item, index) => {
          return (
            <MenuItem key={index} value={item.value}>
              {item.label}
            </MenuItem>
          );
        })}
    </Select>
  );
};

SelectMenu.propTypes = {
  InputId: PropTypes.string,
  InputProps: PropTypes.object,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      label: PropTypes.string,
    })
  ).isRequired,
  multiple: PropTypes.bool,
  MenuProps: PropTypes.object,
  onChange: PropTypes.func,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    ),
  ]).isRequired,
};

export default compose(
  withState('value', 'setValue', ({ value }) => value),
  withHandlers({
    handleChange: ({ setValue, onChange }) => e => {
      const value = e.target.value;
      setValue(value);
      onChange && onChange(value);
    },
  })
)(SelectMenu);
