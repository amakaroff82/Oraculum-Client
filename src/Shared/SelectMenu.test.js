import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SelectMenu } from './SelectMenu';

const stringItems = [
  { value: 'value1', label: 'label1' },
  { value: 'value2', label: 'label2' },
  { value: 'value3', label: 'label3' },
];
const stringValue = stringItems[0].value;

const numberItems = [
  { value: 0, label: 'label1' },
  { value: 1, label: 'label2' },
  { value: 2, label: 'label3' },
];
const numberValue = numberItems[0].value;

const setupShallow = props => {
  return shallow(<SelectMenu {...props} />);
};

describe('<SelectMenu /> component', () => {
  it('renders correctly with string item values', () => {
    const selectMenu = setupShallow({ items: stringItems, value: stringValue });
    expect(toJson(selectMenu)).toMatchSnapshot();
  });

  it('renders correctly with number item values', () => {
    const selectMenu = setupShallow({ items: numberItems, value: numberValue });
    expect(toJson(selectMenu)).toMatchSnapshot();
  });
});
