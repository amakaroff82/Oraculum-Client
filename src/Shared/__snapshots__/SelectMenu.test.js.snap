// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectMenu /> component renders correctly with number item values 1`] = `
<withStyles(Select)
  input={
    <withStyles(Input)
      id="select-input"
    />
  }
  value={0}
>
  <withStyles(MenuItem)
    value={0}
  >
    label1
  </withStyles(MenuItem)>
  <withStyles(MenuItem)
    value={1}
  >
    label2
  </withStyles(MenuItem)>
  <withStyles(MenuItem)
    value={2}
  >
    label3
  </withStyles(MenuItem)>
</withStyles(Select)>
`;

exports[`<SelectMenu /> component renders correctly with string item values 1`] = `
<withStyles(Select)
  input={
    <withStyles(Input)
      id="select-input"
    />
  }
  value="value1"
>
  <withStyles(MenuItem)
    value="value1"
  >
    label1
  </withStyles(MenuItem)>
  <withStyles(MenuItem)
    value="value2"
  >
    label2
  </withStyles(MenuItem)>
  <withStyles(MenuItem)
    value="value3"
  >
    label3
  </withStyles(MenuItem)>
</withStyles(Select)>
`;
